tests/test_algorithm.py

```diff
@@ -3531,6 +3532,42 @@ def handle_data(algo, data):
                                        env=self.env)
         self.check_algo_succeeds(algo, handle_data)
 
+    def test_set_min_leverage(self):
+        # Algo date is not past deadline so algorithm succeeds
+        deadline = pd.Timestamp('2010-01-05', tz='UTC')
+
+        # Set min leverage to 0 so buying one share fails.
+        def handle_data(algo, data):
```

Comment:
I think this `handle_data` definition is sufficient for all the tests in this function.  Seems like it'd be simpler and clearer to just always use this instead of re-defining it for each test?